vizro-core/tests/unit/vizro/models/_components/test_graph.py

```diff
@@ -1,33 +1,108 @@
+        assert graph.figure == standard_px_chart._captured_callable
+        assert graph.actions == []
+
+    @pytest.mark.parametrize("id", ["id_1", "id_2"])
+    def test_create_graph_mandatory_and_optional(self, standard_px_chart, id):
+        graph = vm.Graph(
+            figure=standard_px_chart,
+            id=id,
+            actions=[
+                vm.Action(function=export_data()),
+            ],
+        )
+        actions_chain = graph.actions[0]
+
+        assert graph.id == id
+        assert graph.type == "graph"
+        assert graph.figure == standard_px_chart._captured_callable
+        assert isinstance(actions_chain, ActionsChain)
+
+    def test_mandatory_figure_missing(self):
+        with pytest.raises(ValidationError, match="field required"):
+            vm.Graph()
 
+    def test_failed_graph_with_wrong_figure(self, standard_go_chart):
+        with pytest.raises(ValidationError, match="must provide a valid CapturedCallable object"):
+            vm.Graph(
+                figure=standard_go_chart,
+            )
 
-def test_create_graph(standard_px_chart):
-    graph = vm.Graph(figure=standard_px_chart)
+    @pytest.mark.parametrize("title, expected", [(None, 24), ("Test", None)])
+    def test_title_margin_adjustment(self, gapminder, title, expected):
+        figure = vm.Graph(figure=px.bar(data_frame=gapminder, x="year", y="pop", title=title)).__call__()
 
-    assert hasattr(graph, "id")
-    assert graph.type == "graph"
-    assert graph.figure == standard_px_chart._captured_callable
-    assert graph.actions == []
+        assert figure.layout.margin.t == expected
+        assert figure.layout.template.layout.margin.t == 64
+        assert figure.layout.template.layout.margin.l == 80
+        assert figure.layout.template.layout.margin.b == 64
+        assert figure.layout.template.layout.margin.r == 12
 
 
-def test_failed_graph_with_wrong_figure(standard_go_chart):
-    with pytest.raises(ValidationError, match="must provide a valid CapturedCallable object"):
-        vm.Graph(
-            figure=standard_go_chart,
+class TestProcessFigureDataFrame:
+    def test_process_figure_data_frame_str_df(self, standard_px_chart_with_str_dataframe, gapminder):
+        data_manager["gapminder"] = gapminder
+        graph_with_str_df = vm.Graph(
+            id="text_graph",
+            figure=standard_px_chart_with_str_dataframe,
         )
+        assert data_manager._get_component_data("text_graph").equals(gapminder)
+        assert graph_with_str_df["data_frame"] == "gapminder"
 
+    def test_process_figure_data_frame_df(self, standard_px_chart, gapminder):
+        graph_with_df = vm.Graph(
+            id="text_graph",
+            figure=standard_px_chart,
+        )
+        assert data_manager._get_component_data("text_graph").equals(gapminder)
+        with pytest.raises(KeyError, match="'data_frame'"):
+            graph_with_df.figure["data_frame"]
+
+
+class TestGetitem:
```

Comment:
Is there a reason we have the tests for the `GetItem` in its own class, but the `Call` test in the instantiation? I would probably move them into a class `TestDunderMethodsGraph` or similar. Then its more visually clear